core/src/main/scala/fs2/interop/reactivestreams/StreamUnicastPublisher.scala

```diff
@@ -11,23 +11,31 @@ import scala.concurrent.ExecutionContext
 /** Implementation of an org.reactivestreams.Publisher.
   *
   * This is used to publish elements from an fs2.Stream to a downstream reactivestreams system.
-  * 
+  *
   * @see https://github.com/reactive-streams/reactive-streams-jvm#1-publisher-code
   *
   */
-final class StreamUnicastPublisher[F[_], A](val s: Stream[F, A])(implicit AA: Effect[F], ec: ExecutionContext) extends Publisher[A] {
+final class StreamUnicastPublisher[F[_]: Effect, A](val s: Stream[F, A])
+                                                   (implicit ec: ExecutionContext) extends Publisher[A] {
 
   def subscribe(subscriber: Subscriber[_ >: A]): Unit = {
     nonNull(subscriber)
-    async.unsafeRunAsync(StreamSubscription(subscriber, s).map { sub =>
-      subscriber.onSubscribe(sub)
-    })(_ => IO.unit)
+    async.unsafeRunAsync {
+      StreamSubscription(subscriber, s).map { subscription =>
+        subscriber.onSubscribe(subscription)
+        subscription
+      }
+    } {
+      case Left(err) => IO(err.printStackTrace())
```

Comment:
It's not valid to invoke `onError` unless `onSubscribe` has been called first.  I think this is the best that we can do.